### DISCUSSION TITLE
Why are return values accessible in Cypress if they cannot be used meaningfully?

### DISCUSSION THREAD

**Discussion by Orchi1904** [cypress-io/cypress#28771] [Questions and Help] [answered]
Hello everyone,

I have a question regarding the return values in Cypress. In the Cypress [documentation](https://docs.cypress.io/guides/core-concepts/introduction-to-cypress#Mixing-Async-and-Sync-code), it states:

"You purposely cannot do anything useful with the return value from a command. Commands are enqueued and managed entirely behind the scenes."

What I don't understand is why we are able to retrieve the return values if we cannot perform any useful actions with them? I could not find an answer to that in the documentation.

**Comment by jordanpowell88** [comment, marked as answer]
You can use the return value by passing it from one command to the next (chaining commands) or by using something like `.then()` and and using a closure

**Reply by Orchi1904** [reply]
Thank you for your response. I am aware that this is possible, but in the documentation, it seemed to me that there is one return value that returns instantly and is essentially useless and there is also another return value that is returned after the command has been executed in the queue, which you can use for `.then()` and something like that. Thats why I asked this question, because I got confused by that :D
